src/Admin/Actions.php

```diff
@@ -70,26 +73,170 @@ public function save_admin_settings() {
 			unset( $post_data['roadblock'] );
 
 			if (
-				! empty( $post_data['domain_name'] ) &&
-				! empty( $post_data['custom_domain_prefix'] ) &&
-				! empty( $post_data['self_hosted_domain'] ) &&
-				! empty( $post_data['shared_link'] )
+				! empty( $post_data['is_proxy'] ) &&
+				! empty( $post_data['is_custom_path'] ) &&
+				! empty( $post_data['is_self_hosted_analytics'] ) &&
+				! empty( $post_data['embed_analytics'] ) &&
+				! empty( $post_data['track_administrator'] )
+
 			) {
+
+				if ( empty( $post_data['domain_name'] ) ) {
+					$post_data['domain_name'] = Helpers::get_domain();
+				}
+
+				// Disable embed_analytics if no shared link provided
+				if ( empty( $post_data['shared_link'] ) ) {
+					$post_data['embed_analytics'] = 'false';
+				}
+
+				// Disable is_custom_path if no custom path provided
+				if ( ( $post_data['is_proxy'] === 'false' || empty( $post_data['script_path'] ) || empty( $post_data['event_path'] ) ) ) {
+					$post_data['is_custom_path'] = 'false';
+				}
+
+				// trailing slash on paths settings if filled
+				if ( ! empty( $post_data['script_path'] ) ) {
+					$post_data['script_path'] = trailingslashit( $post_data['script_path'] );
+				}
+				if ( ! empty( $post_data['event_path'] ) ) {
+					$post_data['event_path'] = trailingslashit( $post_data['event_path'] );
+				}
+
 				$status = 'success';
 				update_option( 'plausible_analytics_settings', $post_data );
 				$message = esc_html__( 'Settings saved successfully.', 'plausible-analytics' );
 			} else {
-				$status = 'error';
+				$status  = 'error';
 				$message = esc_html__( 'Something gone a wrong.', 'plausible-analytics' );
 			}
 
+			// Maybe create js files in WordPress root
+			self::maybe_create_js_files();
+
 			// Send response.
 			wp_send_json_success(
 				[
 					'message' => $message,
-					'status' => $status
+					'status'  => $status,
 				]
 			);
 		}
 	}
+
+	/**
+	 * Create JS files if needed
+	 *
+	 * @return void | WP_Error
+	 * @since 1.2.5
+	 *
+	 */
+
+	public static function maybe_create_js_files() {
```

Comment:
What's the plan for:
1. When we add a new script extension (we have a new one coming out in the next few weeks)
2. We make changes to existing script files

Reply:
Open question